Mathlib/Algebra/Algebra/RestrictScalars.lean

```diff
@@ -216,9 +216,8 @@ theorem RestrictScalars.ringEquiv_map_smul (r : R) (x : RestrictScalars R S A) :
 /-- `R ⟶ S` induces `S-Alg ⥤ R-Alg` -/
 instance RestrictScalars.algebra : Algebra R (RestrictScalars R S A) :=
   { (algebraMap S A).comp (algebraMap R S) with
-    smul := (· • ·)
-    commutes' := fun _ _ ↦ Algebra.commutes' _ _
-    smul_def' := fun _ _ ↦ Algebra.smul_def' _ _ }
+    commutes' := fun _ _ ↦ Algebra.commutes' (A := A) _ _
+    smul_def' := fun _ _ ↦ Algebra.smul_def' (A := A) _ _ }
```

Comment:
Why was this change needed?

Reply:
L219: I like deleting these. It is independent of anything else. 
L220-221: I've seen elaboration struggle when the terms get less transparent/more compact. Both here and in other reductions of terms.